extensions/wikia/TemplateClassification/TemplateClassificationController.class.php

```diff
@@ -49,4 +53,78 @@ public function getTemplateClassificationEditForm() {
 
 		$this->setVal( 'templateTypes', $templateTypes );
 	}
+
+	/**
+	 * Classify all templates with given category
+	 *
+	 * @requestParam string category category name
+	 * @requestParam string type template type name
+	 *
+	 * @throws ApiException
+	 * @throws \BadRequestApiException
+	 * @throws \InvalidParameterApiException
+	 * @throws \MWException
+	 * @throws \UnauthorizedException
+	 */
+	public function classifyTemplateByCategory() {
+		$errors = [];
+		$category = $this->request->getVal( 'category', null );
+		$templateType = $this->request->getVal( 'type', null );
+		$userId = $this->wg->User->getId();
+
+		$this->validateRequestForBulkEdit( $category, $templateType );
+
+		$templates = ( new Helper() )->getTemplatesByCategory( $category );
+		$utcs = new UserTemplateClassificationService();
+
+		foreach ( $templates as $templateId => $templateTitle ) {
+			try {
+				$utcs->classifyTemplate( $this->wg->CityId, $templateId, $templateType, 'user', $userId );
+			} catch( ApiException $e ) {
+				$errors[] = Title::newFromText( $templateTitle )->getText();
+			}
+		}
+
+		if ( !empty( $errors ) ) {
+			$this->prepareBulkActionError( $errors, $templates );
+		}
+	}
+
+	private function validateRequestForBulkEdit( $category, $templateType ) {
+		if ( !$this->wg->request->wasPosted() ) {
+			throw new BadRequestApiException();
+		}
+
+		if ( !$this->wg->User->isAllowed( 'template-bulk-classification' )
+			|| !$this->wg->User->matchEditToken( $this->wg->request->getVal( 'editToken' ) )
+		) {
+			throw new UnauthorizedException();
+		}
+
+		if ( empty( $category ) ) {
+			throw new InvalidParameterApiException( 'category' );
+		}
+
+		if ( empty( $templateType ) ) {
+			throw new InvalidParameterApiException( 'template type' );
+		}
+
+		return true;
+	}
+
+	private function prepareBulkActionError( Array $errors, Array $templates ) {
+		$errorsCount = count( $errors );
+
+		if ( $errorsCount === count( $templates ) ) {
+			$errorMessage = wfMessage( 'template-classification-edit-modal-error' )->escaped();
+		} else {
+			$pages = ( new Language() )->listToText( $errors );
```

Comment:
This should use the global language object so that it is in the user's language. Creating a new `Language` object will just use English.